compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs

```diff
@@ -973,6 +974,23 @@ data InterfaceMethod = InterfaceMethod
   }
   deriving (Eq, Data, Generic, NFData, Show)
 
+-- | Template implementation of an interface in the interface declaration.
+data InterfaceCoImplements = InterfaceCoImplements
+  { iciTemplate :: !(Qualified TypeConName)
+  , iciMethods :: !(NM.NameMap InterfaceCoImplementsMethod)
+  }
+  deriving (Eq, Data, Generic, NFData, Show)
+
+-- | Template implementation of an interface's method in the interface declaration.
+data InterfaceCoImplementsMethod = InterfaceCoImplementsMethod
+  { iciMethodName :: !MethodName
+    -- ^ Name of method.
+  , iciMethodExpr :: !Expr
```

Comment:
Is there a reason why you’re going for this as opposed to having a binder for `this` and then having this be of type `mty` where `this` is in scope?

Reply:
no reason, just doing the same as `data TemplateImplementsMethod`. Actually the types are isomorphic but it's probably a good idea to keep them separate

Reply:
Ah you're absolutely right, I found @sofiafaro-da's PR https://github.com/digital-asset/daml/pull/13142/files#diff-d63b60abce4a614e8446abd7ae6c145c6103e7a1d8f813ac023f7bd47f40cc30R968, it seems we just forgot to update the comment. I'll do it while we're here. Thanks for making me check @cocreature!